File: LC 816.cpp
class Solution {
public:
    vector<string> ambiguousCoordinates(string s) {
        int n = s.size() - 2;
        s = s.substr(1, n);
        vector<string>ans;
        auto getPos = [&] (string s) -> vector<string> {
            vector<string> res;
            if (s[0] != '0' || s == "0") res.push_back(s);
            for (int p = 1; p < s.size(); p++) {
                if ((p != 1 && s[0] == '0') || s.back() == '0') continue;
                res.push_back(s.substr(0,p) + "." + s.substr(p));
            }
            return res;
        };
        for (int i = 1; i < n; i++) {
            auto a = s.substr(0, i);
            auto l = getPos(a);
            auto b = s.substr(i);
            auto r = getPos(b);
            for (auto& p : l) {
                for (auto& q : r) {
                    ans.push_back("(" + p + ", " + q + ")");
                }
            }
        }
        return ans;
    }
};

File: 1000/LC 816.go
func ambiguousCoordinates(s string) (ans []string) {
    s = s[1:len(s)-1]
    n := len(s)

    get := func(s string) (res []string) {
        if s[0] != '0' || s == "0" {
            res = append(res, s)
        }
        for i := 1; i < len(s); i++ {
            if (i > 1 && s[0] == '0') || s[len(s)-1] == '0' {
                break
            }
            res = append(res, s[:i] + "." + s[i:])
        }
        return
    }

    for i := 1; i < n; i++ {
        l := get(s[:i])
        r := get(s[i:])
        for _, p := range l {
            for _, q := range r {
                ans = append(ans, "(" + p + ", " + q + ")")
            }
        }
    }

    return
}